src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -707,9 +707,15 @@ private void CompileMethodCleanup()
             _instantiationToJitVisibleInstantiation = null;
 
             _pgoResults.Clear();
+
+            // We need to clear out this cache because the next compilation could actually come up
+            // with a different MethodIL for the same MethodDesc. This happens when we need to replace
+            // a MethodIL with a throw helper.
+            _methodILScopeToHandle.Clear();
         }
 
         private Dictionary<object, IntPtr> _objectToHandle = new Dictionary<object, IntPtr>(new JitObjectComparer());
+        private Dictionary<MethodDesc, IntPtr> _methodILScopeToHandle = new Dictionary<MethodDesc, IntPtr>(new JitObjectComparer());
```

Comment:
Just for my education - this is a per-method cache, as in, we're going to clear it after each method compilation. And JIT doesn't maintain any state between methods either, right?

Reply:
Yes, it's per method and there's not supposed to be extra state on the JIT side either.